js/bundle.go

```diff
@@ -262,26 +263,31 @@ func (b *Bundle) Instantiate(logger logrus.FieldLogger, vuID uint64) (*BundleIns
 	}
 
 	rt := vuImpl.runtime
+	pgm := init.programs[b.Filename.String()]
 	bi := &BundleInstance{
 		Runtime:      rt,
 		exports:      make(map[string]goja.Callable),
 		env:          b.RuntimeOptions.Env,
 		moduleVUImpl: vuImpl,
+		pgm:          pgm,
 	}
 
 	// Grab any exported functions that could be executed. These were
 	// already pre-validated in cmd.validateScenarioConfig(), just get them here.
-	exports := rt.Get("exports").ToObject(rt)
+	exports := pgm.module.Get("exports").ToObject(rt)
 	for k := range b.exports {
 		fn, _ := goja.AssertFunction(exports.Get(k))
 		bi.exports[k] = fn
 	}
 
-	jsOptions := rt.Get("options")
+	jsOptions := exports.Get("options")
 	var jsOptionsObj *goja.Object
```

Comment:
I think it's possible - not certain how well it will play once we have ESM, but it should work :shrug:

Reply:
When I tried when I implemented `test.options` the issue was in the case of `export const` where IIRC it isn't compatible with dynamic solutions.

Reply:
https://github.com/grafana/k6/issues/2601 please comment if I've missed something

Reply:
Hmm this in practice will not work ... all that well even with just this cchanges as we already can *not* change the original `options`. This here only sets `exports.options` in case they are empty. 

even before that again we were just setting something *globally*. 

This won't work with ESM as well as in that case just as now we don't have access to change waht `options` is in hte scope of the module.

Reply:
Can't we `exports.Set("options", jsOptionsObj)` in both cases? :thinking:

Reply:
Not really. In the case of commonjs transpiled by babel the actual code is something like 

```
let options = {...}
exports.options = options;

// and then later it's accessed as `options` 
options.scenario.....
```
In this case we can resset `exports.options` but nothing will be using it. What we have done so far (and what this PR changes) is basically setting `globalThis.options` which just so happens to be the same as having `let options` in the global scope. 

In ESM 
```
export let options ={}
```
is more or less 

```
let options = {}
export options;
```
And then we get access to `options` and can set it's fields (just as before) but AFAIK there is no way to just change what `options` is supposed to be.

You can think of it as modules are supposed to encapsulate stuff so that the outside world can't mangle them and what you want is basically what modules are kind of trying to prevent. (They also prevent spilling stuff from inside modules in the global scope which is the other side of the coin - arguably the one that is more commonly cited) 

So it seems  like this isn't really a thing we can do :shrug:

Reply:
Got it, I think... Then this whole logic of updating the script `options` with the consolidated values won't work at all? :confused:

Reply:
Or maybe it will, since we're going item by item and updating individual keys of the object? :thinking: In which case, we might be able to add the usage warnings at the key level?

Reply:
This, I mean: https://github.com/grafana/k6/blob/6b77bfc291f17f6de340057105097c34ae312971/js/bundle.go#L297-L299

Reply:
This should work for all varitants :thinking: